integrationtests/gitjob/git/git_test.go

```diff
@@ -62,309 +65,373 @@ var (
 	container               testcontainers.Container
 )
 
-func TestMain(m *testing.M) {
-	teardown := setupSuite()
-	code := m.Run()
-	teardown(code)
-	os.Exit(code)
-}
-
-func setupSuite() func(code int) {
+var _ = BeforeSuite(func() {
 	ctx := context.Background()
 	var err error
-	t := &testing.T{}
-	container, url, err = createGogsContainer(ctx, createTempFolder(t))
-	require.NoError(t, err, "creating gogs container failed")
-
-	return func(code int) {
-		terminateContainer(ctx, t, container, code)
-	}
-}
+	tmpDir, err := os.MkdirTemp("", "gogs")
+	Expect(err).NotTo(HaveOccurred())
+
+	container, url, err = createGogsContainer(ctx, tmpDir)
+	Expect(err).NotTo(HaveOccurred())
+})
+
+var _ = AfterSuite(func() {
+	if container != nil {
+		ctx := context.Background()
+		err := container.Terminate(ctx)
+		Expect(err).NotTo(HaveOccurred())
+	}
+})
+
+var _ = Describe("Git Fetch", func() {
+	Describe("LatestCommit NoAuth", func() {
+		var ctx context.Context
+		var ctlr *gomock.Controller
+
+		BeforeEach(func() {
+			ctx = context.Background()
+			ctlr = gomock.NewController(GinkgoT())
+			config.Set(&config.Config{})
+		})
 
-func TestLatestCommit_NoAuth(t *testing.T) {
-	ctlr := gomock.NewController(t)
-	defer ctlr.Finish()
-	ctx := context.Background()
+		AfterEach(func() {
+			ctlr.Finish()
+		})
 
-	tests := map[string]struct {
-		gitrepo        *v1alpha1.GitRepo
-		expectedCommit string
-		expectedErr    error
-	}{
-		"public repo": {
-			gitrepo: &v1alpha1.GitRepo{
+		It("public repo", func() {
```

Comment:
I prefer block statements reading like sentences, such as:
```suggestion
		It("fetches the latest commit from a public repo", func() {
```